BaseTools/Conf/build_rule.template

```diff
@@ -311,6 +311,10 @@
     <OutputFile>
         $(DEBUG_DIR)(+)$(MODULE_NAME).dll
 
+    <Command.MSFT>
+        "$(DLINK)" /OUT:${dst} $(DLINK_FLAGS) $(DLINK2_FLAGS) $(DLINK_XIPFLAGS) $(DLINK_SPATH) @$(STATIC_LIBRARY_FILES_LIST)
+        "$(DLINK)" /OUT:${dst} $(DLINK_FLAGS) $(DLINK_XIPFLAGS) $(DLINK_SPATH) @$(STATIC_LIBRARY_FILES_LIST)
+

```

Comment:
Why are there 2 DLINK steps here to the same output file?  This would seem to drop the use of  DLINK2_FLAGS

Reply:
@mdkinney , this was added for `/WHOLEARCHIVE` 7 years ago: https://bugzilla.tianocore.org/show_bug.cgi?id=582. It does create a temporary linked file, but doesn't want to have the large size impact.

Reply:
Thanks for the link to BZ.  I remember this now.  It is unfortunate that DLINK2_FLAGS is used for different reasons for different toolchains.  Makes it a bit confusing.  

I see that the line to add /WHOLEARCHIVE to VS2022 for IA32/X64 was commented out and then removed.  Is /WHOLEARCHIVE set in DLINK2_FLAGS for IA32/X64 elsewhere?

Reply:
It is set for VS2019 and earlier: https://github.com/tianocore/edk2/blob/e53cf2412a34c9d3a9805fe8dfd2b51f5adfff21/BaseTools/Conf/tools_def.template#L612. Otherwise DLINK2_FLAGS are not used on VS.

Reply:
Why is this not used for VS2022?  Detecting multiply defined symbols is an important CI check.

Reply:
There was some history here on this causing platform build issues. I ran a CI build with it locally and it looks okay as expected, so I will add it back in a separate commit in this PR.